lib/api/apiUtils/object/BackendInfo.js

```diff
@@ -0,0 +1,144 @@
+import config from '../../../Config';
+import utils from '../../../utils';
+
+export class BackendInfo {
+    /**
+    * Represents the info necessary to evaluate which data backend to use
+    * on a data put call.
+    * @constructor
+    * @param {string | undefined} objectLocationConstraint - location constraint
+    * for object based on user meta header
+    * @param {string | undefined } bucketLocationConstraint - location
+    * constraint for bucket based on bucket metadata
+    * @param {string} requestEndpoint - endpoint to which request was made
+    */
+    constructor(objectLocationConstraint, bucketLocationConstraint,
+        requestEndpoint) {
+        this._objectLocationConstraint = objectLocationConstraint;
+        this._bucketLocationConstraint = bucketLocationConstraint;
+        this._requestEndpoint = requestEndpoint;
+        return this;
+    }
+
+    /**
+     * validate proposed location constraint against config
+     * @param {string | undefined} locationConstraint - value of user
+     * metadata location constraint header or bucket location constraint
+     * @param {object} log - werelogs logger
+     * @return {boolean} - true if valid, false if not
+     */
+    static isValidLocationConstraint(locationConstraint, log) {
+        if (config.regions &&
+            Object.keys(config.regions).indexOf(locationConstraint) < 0) {
+            log.trace('proposed locationConstraint is invalid',
+                { locationConstraint });
+            return false;
+        } else if (!config.regions && config.locationConstraints &&
+            Object.keys(config.locationConstraints)
+            .indexOf(locationConstraint) < 0) {
+            log.trace('proposed locationConstraint is invalid',
+                { locationConstraint });
+            return false;
+        }
+        return true;
+    }
+
+    /**
+     * validate requestEndpoint against config
+     * if there is a mismatch between the request endpoint and what is in
+     * the config, this could cause a problem for setting the backend location
+     * for storing data
+     * @param {string} requestEndpoint - request endpoint
+     * @param {object} log - werelogs logger
+     * @return {boolean} - true if valid, false if not
+     */
+    static isValidRequestEndpoint(requestEndpoint, log) {
+        if (utils.getAllEndpoints().indexOf(requestEndpoint) < 0) {
```

Comment:
this should have been Object.keys(config.restEndpoints).

We want to make sure that the requestEndpoint matches an entry in the dictionary of restEndpoints